ApiCheck/Loader/AssemblyLoader.cs

```diff
@@ -15,21 +15,21 @@ public void Dispose()
     {
       AppDomain.CurrentDomain.ReflectionOnlyAssemblyResolve -= CurrentDomainOnReflectionOnlyAssemblyResolve;
     }
-    
+
     private Assembly CurrentDomainOnReflectionOnlyAssemblyResolve(object sender, ResolveEventArgs args)
     {
       AssemblyName assemblyName = new AssemblyName(args.Name);
       string path = Path.Combine(Path.GetDirectoryName(args.RequestingAssembly.Location), assemblyName.Name + ".dll");
       if (File.Exists(path))
       {
-        return Assembly.ReflectionOnlyLoadFrom(path);
+        return Assembly.LoadFile(path);
       }
       return Assembly.ReflectionOnlyLoad(AppDomain.CurrentDomain.ApplyPolicy(args.Name));
     }
 
     public Assembly ReflectionOnlyLoad(string path)
     {
-      return Assembly.ReflectionOnlyLoadFrom(path);
+      return Assembly.LoadFile(Path.GetFullPath(path));
```

Comment:
By using `Assembly.LoadFile` the assembly will be loaded in the AppDomain for code execution which has stricter rules than `ReflectionOnlyLoadFrom`. For example dependencies will also need to be loaded. However, we only need to load the assembly for inspecting it's contents. As an alternative to reflection we can have a look at [MetadataLoadContext](https://learn.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext)